Docs/dev/how_to_add_a_new_sensor.md

```diff
@@ -0,0 +1,569 @@
+<h1>How to add a new sensor</h1>
+
+This tutorial explains the basics for adding a new sensor to CARLA. It provides
+the necessary steps to implement a sensor in Unreal Engine 4 (UE4) and expose
+its data via CARLA's Python API. We'll follow all the steps by creating a new
+sensor as an example.
+
+## Prerequisites
+
+In order to implement a new sensor, you'll need to compile CARLA source code,
+for detailed instructions on how to achieve this see
+[Building from source](../building_from_source.md).
+
+This tutorial also assumes the reader is fluent in C++ programming.
+
+## Introduction
+
+Sensors in CARLA are a special type of actor that produce a stream of data. Some
+sensors produce data continuously, every time the sensor is updated, other
+produce data only after certain events. For instance, a camera produces an image
+on every update, but a collision sensor is only triggered in the event of a
+collision.
+
+Although most sensors compute their measurements in the server side (UE4), it's
+worth noticing that some sensors run in the client-side only. An example of such
+sensor is the GNSS, it computes the simulated geo-location in the client-side
+based on its 3D location. For further details see
+[Appendix: Client-side sensors](#appendix-client-side-sensors).
+
+In this tutorial, we'll be focusing on server-side sensors.
+
+In order to have a sensor running inside UE4 sending data all the way to a
+Python client, we need to cover the whole communication pipeline.
+
+![Communication pipeline](../img/pipeline.png)
+
+Thus we'll need the following classes covering the different steps of the
+pipeline
+
+  * **Sensor actor**<br>
+    Actor in charge of measuring and/or simulating data. Running in Carla plugin
+    using UE4 framework. Accessible by the user as Sensor actor.
+
+  * **Serializer**<br>
+    Object containing methods for serializing and deserializing the data
+    generated by the sensor. Running in LibCarla, both server and client.
+
+  * **Sensor data**<br>
+    Object representing the data generated by the sensor. This is the object
+    that will be passed to the final user, both in C++ and Python APIs.
+
+!!! note
+    To ensure best performance, sensors are registered and dispatched using a
+    sort of "compile-time plugin system" based on template meta-programming.
+    Most likely, the code won't compile until all the pieces are present.
+
+## Creating a new sensor
+
+[**Full source code here.**](https://gist.github.com/nsubiron/011fd1b9767cd441b1d8467dc11e00f9)
```

Comment:
Maybe source code for example could be in this repository under clearly named folder structure?